main/src/cgeo/geocaching/storage/DataStore.java

```diff
@@ -477,7 +477,7 @@ private static File databasePath(final boolean internal) {
     }
 
     @NonNull
-    private static File databasePath() {
+    public static File databasePath() {
```

Comment:
You could leave the methods private by moving the code of appendDirectory and appendDatabase into LocalStorage and DataStore (and probably rename them to something like LocalStorage.appendDirectorySystemInformation(...), appendDatabaseSystemInformation).

However, I'm aware that both alternatives have drawbacks (one needing public access, the other smearing functionality of the system information over multiple classes), so I can't really say what to prefer due to what reasoning.